.github/workflows/ubsan.yml

```diff
@@ -0,0 +1,152 @@
+name: UBSAN build tests
+
+on: [ push ]
+
+# INSTRUCTIONS for setting up a self-hosted runner:
+# Install Ubuntu 20.04
+# Install base dependencies
+#   add-apt-repository ppa:canonical-server/server-backports
+#   apt update
+#   apt install build-essential make cmake clang-format-11 gcc-10 g++-10 libstdc++-10-dev libgmp-dev ccache libmpfr-dev libmpc-dev ninja-build e2fsprogs qemu-utils qemu-system-i386 binaryen
+# Force usage of GCC 10
+#   update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 100 --slave /usr/bin/g++ g++ /usr/bin/g++-10
+# Create a new user account named 'runner'
+#   adduser runner
+# Give it password-less sudo capabilities by adding the following line to /etc/sudoers:
+#   runner ALL=(ALL) NOPASSWD:ALL
+# Add it to the kvm access group:
+#   adduser runner kvm
+# Switch to the new user and then create a workspace folder:
+#   mkdir actions-runner && cd actions-runner
+# Download the latest version of actions-runner-linux-x64 from https://github.com/rust-lang/gha-runner/releases
+#   curl -o actions-runner-linux-x64-X.X.X.tar.gz -L https://github.com/rust-lang/gha-runner/releases/download/vX.X.X-rust1/actions-runner-linux-x64-X.X.X-rust1.tar.gz
+# Extract the tar archive
+#   tar xzf ./actions-runner-linux-x64-X.X.X.tar.gz
+# Link the runner
+#   ./config.sh --url https://github.com/SerenityOS/serenity --token INSERT_SECRET_TOKEN_HERE
+# Configure the runner to protect against malicious PRs by adding the following line to .env:
+#   RUST_WHITELISTED_EVENT_NAME=push
+# Install the runner as a service
+#   sudo ./svc.sh install
+# Start the runner
+#   sudo ./svc.sh start
```

Comment:
Let's put all of this in a separate markdown document (probably in Documentation/) instead of hiding it in the workflow config file :)
We can still reference that from here.

Reply:
sure thing!